### PR TITLE
Use new configuration model

### PR DESCRIPTION
**Pull request by atstojanov**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/cherrytwist/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/cherrytwist/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/cherrytwist/Coordination/blob/master/LICENSE 
     and the contributor license agreement: tba

### CODE REVIEW

**Review by techsmyth — CHANGES_REQUESTED** [review]
Names to be updated to match those from last commit on server side:
- AadConfig ==> AadAuthProviderConfig
- ConfigUnion ==> AuthProviderConfigUnion

File: src/hooks/useAuthentication.ts [hunk 1]
```diff
@@ -69,9 +69,14 @@ const refreshToken = async (msalApp?: PublicClientApplication, aadConfig?: AadCo
 };
 
 export const useAuthentication = (): UseAuthenticationResult => {
-  const { loading: configLoading, aadConfig } = useContext(configContext);
+  const { loading: configLoading, config } = useContext(configContext);
+
+  const aadConfig = useMemo(() => {
+    return config.authentication.providers.find(x => x.config.__typename === 'AadConfig')?.config as AadConfig;
```
Comment (valentinyanakiev) [inline src/hooks/useAuthentication.ts:75]: I think with the latest changes this is AadAuthProviderConfig

File: src/graphql/config.ts [hunk 2]
```diff
@@ -1,36 +1,48 @@
...
+    configuration {
+      authentication {
+        enabled
+        providers {
+          name
+          label
+          icon
+          config {
+            __typename
+            ... on AadConfig {
```
Comment (valentinyanakiev) [inline src/graphql/config.ts:14]: AadAuthProviderConfig

File: src/generated/graphql.tsx [hunk 3]
```diff
@@ -3920,6 +3978,77 @@ export function useOrganizationCardLazyQuery(
...
+    configuration {
+      authentication {
+        enabled
+        providers {
+          name
+          label
+          icon
+          config {
+            __typename
+            ... on AadConfig {
```
Comment (valentinyanakiev) [inline src/generated/graphql.tsx:3992]: AadAuthProviderConfig, you need to update the schema against the latest definition

**Review by valentinyanakiev — CHANGES_REQUESTED** [review]
Update against the latest schema - this configuration shouldn't work with it. It's AadAuthProviderConfig instead of AadConfig.

**Review by valentinyanakiev — APPROVED** [review]

**Pull request merged**